tools/core_packages.txt

```diff
@@ -81,6 +81,8 @@ ruby
 sed
 slang
 sqlite
+tcl
+tk
```

Comment:
On a fresh install, I don't see the following:
```
groff
libunbound
libxdmcp
pcre2
trousers
uchardet
xorg_proto
```
Can we remove these packages?

Reply:
Also, I attempted to install python3 after a fresh install and got the following NEW dependencies.  These will need to be added to `tools/core_packages.txt`.
````
$ crew install python3
python3: Python is a programming language that lets you work quickly and integrate systems more effectively.
https://www.python.org/
Version: 3.9.4
License: PSF-2.0
The following packages also need to be installed: 
fontconfig gperf freetype_sub libpng jsonc util_linux libcap_ng libtinfo linux_pam libdb pcre2 libeconf graphite freetype harfbuzz gobject_introspection glib pcre shared_mime_info six pygments libx11 xorg_proto libxcb xcb_proto libxau pthread_stubs libxtrans libxext libxft libxrender libxss tcl tk xorg_lib libfontenc libxinerama libxdamage libxfixes libxtst libxi libpciaccess libxkbfile libxshmfence libxcursor libdmx libfs libice libbsd libsm libxaw libxmu libxt libxpm libxfont2 fop jdk8 libxrandr libxres libxv libxvmc libxxf86dga libxxf86vm libxcomposite 
Do you agree? [Y/n]
```
Not sure why we need all the X window dependencies for python3.  Can you explain?

Reply:
After combining the 2 package lists from above, I got the complete list of changes needed as shown below:
```
+fontconfig
+fop
+freetype
+freetype_sub
+glib
+gobject_introspection
+gperf
+graphite
-groff
+harfbuzz
+jdk8
+jsonc
+libbsd
+libcap_ng
+libdb
+libdmx
+libeconf
+libfontenc
+libfs
+libice
+libpciaccess
+libpng
+libsm
+libtinfo
-libunbound
+libx11
+libxau
+libxaw
+libxcb
+libxcomposite
+libxcursor
+libxdamage
-libxdmcp
+libxext
+libxfixes
+libxfont2
+libxft
+libxi
+libxinerama
+libxkbfile
+libxmu
+libxpm
+libxrandr
+libxrender
+libxres
+libxshmfence
+libxss
+libxt
+libxtrans
+libxtst
+libxv
+libxvmc
+libxxf86dga
+libxxf86vm
+linux_pam
+pcre
+pthread_stubs
+pygments
+shared_mime_info
+six
+tcl
+tk
-trousers
-uchardet
+util_linux
+xcb_proto
+xorg_lib
+xorg_proto
```
If we can figure out a way to NOT have all the X window libraries and jdk8, that would be preferrable.

Reply:
It's from `tk`:
_tkinter.cpython-39-x86_64-linux-gnu.so:libXrender.so.1

Hmm. Let me rebuild this w/o tk.

Reply:
I'll add tkinter as its own package :)

Reply:
@saltedcoffii Thanks. I'm rebuilding with `--disable-tcl --disable-tk`

Reply:
Hmm. Still seeing this:
```
  depends_on 'fontconfig'
  depends_on 'freetype'
  depends_on 'glib'
  depends_on 'graphite'
  depends_on 'harfbuzz'
  depends_on 'libdb'
  depends_on 'libpng'
  depends_on 'libx11'
  depends_on 'libxau'
  depends_on 'libxcb'
  depends_on 'libxext'
  depends_on 'libxft'
  depends_on 'libxrender'
  depends_on 'libxss'
  depends_on 'pcre'
  depends_on 'tcl'
  depends_on 'tk'
```

Maybe I need `--with-x=no` ?

Reply:
That doesn't work either. I would suggest INTENTIONALLY avoiding including tcl/tk in base, and maybe just create a dummy package for tkinter which includes those dependencies. (and tcl/tk can be left as build dependencies for python3.)

Reply:
It looks like `libdb` needs to be added to `tools/core_packages.txt`.  It'd be nice to remove the other dependencies also.

Reply:
Added `libdb` to core, and removed the other deps.